plugins/out_s3/s3.c

```diff
@@ -1122,8 +1163,12 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         m_upload->upload_errors += 1;
         /* re-add chunk to list */
         if (chunk) {
-            s3_store_file_unlock(chunk);
             chunk->failures += 1;
+            int less_than_limit = FLB_TRUE;
```

Comment:
variables should always be defined at top level of functions: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md